x/evm/types/types.go

```diff
@@ -959,6 +960,52 @@ func NewCommandBatchMetadata(blockHeight int64, chainID sdk.Int, keyID multisig.
 	}, nil
 }
 
+func (m CommandBatchMetadata) ValidateBasic() error {
+	if m.Status == BatchNonExistent {
+		return errors.New("batch does not exist")
+	}
+
+	if len(m.ID) != 32 {
+		return errors.New("batch ID must be of length 32")
+	}
+
+	if len(m.CommandIDs) == 0 {
+		return errors.New("command IDs must not be empty")
+	}
+
+	if len(m.Data) == 0 {
```

Comment:
We can technically also validate `.Data` here but it might be an overkill.

Reply:
how, we don't know anything about data, do we?